### PR TITLE
#161967090 Enable social authentication

### PR DESCRIPTION
**Pull request by bettblake08**
**What does this PR do?**
-  Enable users to log into the app using their favorite social media platform

**Description of tasks to be completed?**
- Create a social media login component
- Create call-to-action pages for a successful and failed attempts at social authentication

**How should this be manually tested?**
- Clone the repo and cd into the project root folder
- Switch to the relevant branch `git checkout ft-social-auth-161967090`
- Run `npm install` to update your packages.
- Start the server with `npm start`
- In your favorite browser, navigate to `http://localhost:3000`
- To test the call to actions, in your browser, navigate to:
  `http://localhost:3000/social/auth?success=true&username=johndoe`

**What are the relevant pivotal tracker stories?**
- [#161967090](https://www.pivotaltracker.com/story/show/161967090)

**Screenshots**
image
image

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/containers/headers/generalHeader.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { Component } from "react";
+import GeneralHeaderView from "../../views/headers/GeneralHeader";
+
+export default class GeneralHeaderComponent extends Component {
+  constructor(props) {
+    super(props);
```
Comment (denomash) [inline src/components/containers/headers/generalHeader.js]: @bettblake08 Can this be made to be a stateless function

**Review by denomash — CHANGES_REQUESTED** [review]

File: src/components/containers/headers/userHeader.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { connect } from "react-redux";
+import UserHeaderView from "../../views/headers/UserHeader";
+
+class UserHeader extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+    };
+  }
```
Comment (denomash) [inline src/components/containers/headers/userHeader.js]: I don't think we need `this.state `object, since it's not used in this class based function

**Review by denomash — CHANGES_REQUESTED** [review]

**Review by jillo-abdullahi — COMMENTED** [review]
The social logins do not render the home page with the username after a successful login. I can see the username set in the URL but not on the header. Please check for all social login sites.

File: src/components/views/SocialAuthenticationForm.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      </Container>
+    </div>
+  );
+}
+
+SocialAuthentication.propTypes = {
+  parent: PropTypes.object
+};
+
+SocialAuthentication.defaultProps = {
```
Comment (Bryan-Cee) [inline src/components/views/SocialAuthenticationForm.js:75]: Please remove this repetition

File: src/setupTests.js [hunk 4]
```diff
@@ -1,4 +1,29 @@
-import { configure } from "enzyme";
+/* eslint-disable import/prefer-default-export */
+import React from "react";
+import { configure, mount } from "enzyme";
 import Adapter from "enzyme-adapter-react-16";
+import { Provider } from "react-redux";
+import { ConnectedRouter } from "connected-react-router";
+
+// import setupBase from "./setupBase";
```
Comment (Bryan-Cee) [inline src/setupTests.js:8]: Please remove this line of commented code if it is not in use

**Review by Bryan-Cee — CHANGES_REQUESTED** [review]

File: src/components/views/SocialAuthenticationForm.js (hunk 3 above)
Comment (bettblake08) [reply src/components/views/SocialAuthenticationForm.js:75]: defaultProps and prop-types are different. The linting rules state to have default props for any prop that is not required.

File: src/setupTests.js (hunk 4 above)
Comment (bettblake08) [reply src/setupTests.js:8]: My apologies. This line is a piece of code I will want to try to implement soon for testing after this week's sprint. It was intentional.

**Comment by jillo-abdullahi** [comment]
Social logins working fine now. 👍

**Review by davishooly — APPROVED** [review]

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Pull request merged**